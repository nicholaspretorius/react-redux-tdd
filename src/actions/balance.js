import * as constants from "./constants";

export function setBalance(balance) {
  return {
    type: constants.SET_BALANCE,
    balance
  };
}
