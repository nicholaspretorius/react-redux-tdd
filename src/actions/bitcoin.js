import * as constants from "./constants";

function fetchBitcoin(bitcoin) {
  return {
    type: constants.FETCH_BITCOIN,
    bitcoin
  };
}

export function handleFetchBitcoin() {
  return dispatch => {
    return fetch("https://api.coindesk.com/v1/bpi/currentprice.json")
      .then(response => {
        return response.json();
      })
      .then(json => {
        dispatch(fetchBitcoin(json));
      });
  };
}
