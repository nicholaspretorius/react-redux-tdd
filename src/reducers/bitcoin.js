import * as constants from "./../actions/constants";

const bitcoin = (state = {}, action) => {
  switch (action.type) {
    case constants.FETCH_BITCOIN:
      return action.bitcoin;
    default:
      return state;
  }
};

export default bitcoin;
