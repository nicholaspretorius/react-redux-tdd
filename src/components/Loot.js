import React, { Component } from "react";
import { connect } from "react-redux";

import { fetchBitcoin } from "./../actions/bitcoin";

export class Loot extends Component {
  state = {};

  componentDidMount() {
    this.props.fetchBitcoin();
  }

  render() {
    return <div>Loot</div>;
  }
}

export default connect(
  state => state,
  { fetchBitcoin }
)(Loot);
