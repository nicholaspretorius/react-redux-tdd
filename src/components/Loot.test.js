import React from "react";
import { mount, shallow } from "enzyme";

import { Loot } from "./Loot";
import { fetchBitcoin } from "./../actions/bitcoin";

describe("Loot", () => {
  const mockFetchBitcoin = jest.fn();
  const props = { balance: 10, bitcoin: {} };
  let loot = shallow(<Loot {...props} />);

  it("should render the Loot component properly", () => {
    expect(loot).toMatchSnapshot();
  });

  describe("when mounted", () => {
    beforeEach(() => {
      props.fetchBitcoin = mockFetchBitcoin;
      loot = mount(<Loot {...props} />);
    });

    it("dispatches the `fetchBitcoin()` method it receives from props", () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });
  });
});
