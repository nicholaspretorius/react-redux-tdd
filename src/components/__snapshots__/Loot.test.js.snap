// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot should render the Loot component properly 1`] = `ShallowWrapper {}`;
